src/transport_ofi.c

```diff
@@ -1688,16 +1695,16 @@ int shmem_transport_fini(void)
     /* Free all shareable contexts.  This performs a quiet on each context,
      * ensuring all operations have completed before proceeding with shutdown. */
 
-    for (size_t i = 0; i < shmem_transport_ofi_contexts_len; ++i) {
-        if (shmem_transport_ofi_contexts[i]) {
-            if (shmem_transport_ofi_is_private(shmem_transport_ofi_contexts[i]->options))
+    for (size_t i = 0; i < shmem_internal_team_world.contexts_len; ++i) {
+        if (shmem_internal_team_world.contexts[i]) {
+            if (shmem_transport_ofi_is_private(shmem_internal_team_world.contexts[i]->options))
                 RAISE_WARN_MSG("Shutting down with unfreed private context (%zu)\n", i);
-            shmem_transport_quiet(shmem_transport_ofi_contexts[i]);
-            shmem_transport_ctx_destroy(shmem_transport_ofi_contexts[i]);
+            shmem_transport_quiet(shmem_internal_team_world.contexts[i]);
+            shmem_transport_ctx_destroy(shmem_internal_team_world.contexts[i]);
         }
     }
 
-    if (shmem_transport_ofi_contexts) free(shmem_transport_ofi_contexts);
+    if (shmem_internal_team_world.contexts) free(shmem_internal_team_world.contexts);
 
```

Comment:
Missing destroy of shared team.  Can we call `shmem_internal_team_destroy` rather than doing this manually here?

Reply:
I'm a little confused here.  Don't we destroy all teams in `shmem_internal_team_pool`, which should include the pre-defined teams during `shmem_internal_team_fini`?  So do we even need this `free` here?

Reply:
It certainly looks like `shmem_internal_team_world` is freed twice.  In general, I'm in favor of not having special cases.  Hopefully transport_fini doesn't need the default context.

Reply:
This one might be a little tricky since the spec says it is an error to destroy a predefined team.  Would it work to check for the predefined teams in `teams_c.c4` so we can destroy predefined teams internally with `shmem_internal_team_destroy`?

Reply:
I also have a leftover STX in the unit tests, `shmem_ctx_pipelined_reduce` and `many-ctx`, when I try to use `shmem_internal_team_destroy` on the predefined teams during finalization.  I'm stumped on this one at the moment...
```
[0000] WARN:  ../../src/transport_ofi.c:1710: shmem_transport_fini
[0000]        Closing a shared STX (0) with nonzero ref. count (1)
[0000] ERROR: ../../src/transport_ofi.c:1739: shmem_transport_fini
[0000]        AV close failed (Operation now in progress)
```

Reply:
@jdinan please review https://github.com/Sandia-OpenSHMEM/SOS/pull/886/commits/57d414bc6c22373a3b727bf50b390950d6bef462